lib/mongo/crypt/handle.rb

```diff
@@ -108,16 +108,7 @@ def initialize_mongocrypt
       def raise_from_status
         Status.with_status do |status|
           Binding.mongocrypt_status(@mongocrypt, status.ref)
-
-          error = case status.label
-          when :error_kms
-            # There is currently no test for this code path
-            Error::CryptKmsError.new(status.code, status.message)
-          when :error_client
-            Error::CryptClientError.new(status.code, status.message)
-          end
-
-          raise error
+          status.raise_crypt_error
```

Comment:
I'm not sure if there's a better pattern I could use to accomplish this -- maybe
```
raise Mongo::Error::CryptError.from_status(status)
```
Open to opinions here.